bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/navigation/graph/RootNavGraph.kt

```diff
@@ -0,0 +1,59 @@
+package network.bisq.mobile.presentation.ui.navigation.graph
+
+import androidx.compose.animation.AnimatedContentTransitionScope
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import network.bisq.mobile.presentation.ui.navigation.*
+import network.bisq.mobile.presentation.ui.screens.*
+import network.bisq.mobile.presentation.ui.theme.BisqTheme
+
+@Composable
+fun RootNavGraph(

```

Comment:
This one is more of a question than a comment. 

So I can see this is a UI component that allows us to navigate the screens.

When we start implementing business logic this rootnavgraph should have a presenter parameter that decides where to go based on conditions?

I guess I want to make sure with this comment that we can easily refactor to make the presenter decide where to go. 

Typical use case e.g. for the `xClients` -> we only show the "setup trusted node" screen if there isn't any node configured yet

Makes sense? let me know your thoughts

Reply:
So `RootNavGraph` just defines which routes are associated with which Screens/Views.

Like
 /splash -> SplashScreen
 /onboarding -> OnboardingScreen
 /etc.,
 
But this doesn't define the triggers for navigation. So conditions can't be added here.

Let's say,
After SplashScreen,
 - `androidNode` should go to different screen and `xClients` needs to go different screen.
 - Also there is another condition. On first time app startup, it's a different screenflow. But on further app launches, different screen flow.
 - So there are 2 variables here. 1. buildMode 2. firstTimeStartup.
 - Based on this conditions, SplashPresenter will decide, after splash to which route app should go to.
 - RootNavGraph, just maps that route to a specific screen.
 
 Does this makes sense?

Reply:
would be good to add a small readme section explaining the routing if you would like (at least point to the important classes)

Reply:
excellent!!!